kotlinx-coroutines-core/common/src/CompletableDeferred.kt

```diff
@@ -45,6 +45,17 @@ public interface CompletableDeferred<T> : Deferred<T> {
     public fun completeExceptionally(exception: Throwable): Boolean
 }
 
+/**
+ * Completes this deferred value with the value or exception in the given [result]. Returns `true` if this deferred
+ * was completed as a result of this invocation and `false` otherwise (if it was already completed).
+ *
+ * Subsequent invocations of this function have no effect and always produce `false`.
+ *
+ * This function transitions this deferred in the same ways described by [CompletableDeferred.complete] and
+ * [CompletableDeferred.completeExceptionally].
+ */
+public fun <T> CompletableDeferred<T>.completeWith(result: Result<T>) = result.fold({ complete(it) }, { completeExceptionally(it) })
```

Comment:
That looks great. To make it directly mergeable, please add the following annotation and comment:

```
@ExperimentalCoroutinesApi // since 1.3.1, tentatively until 1.4.0
```

Also, you need to run `./gradlew clean :binary-compatibility-validator:test -Poverwrite.output=true` and add resulting change to public API declarations data.

Reply:
Should that be 1.3.2? 1.3.1 is already out, right?

Reply:
Yes, it has to be 1.3.2 now

Reply:
Thanks, done. It would be helpful if that step was described in the contributions section of the README.

Reply:
This step is non-deterministic :)

Reply:
Sorry, I just meant regenerating the binary compatibility files.